src/app/settings/admin/dynamic-datacenters/datacenter-data-dialog/component.ts

```diff
@@ -74,6 +79,17 @@ export class DatacenterDataDialogComponent implements OnInit, OnDestroy {
     private readonly _notificationService: NotificationService
   ) {}
 
+  get label(): string {
```

Comment:
This method is also duplicated in multiple files so may be we can also create generic method in shared enum file (and pass `mode` and `message` as arguments). Wdyt?

Reply:
im not sure if i get your point correctly but if i need to send the mode and the returning message for that mode then why we need a generic method to get back the same massage ? or what exactly do you mean ?

Reply:
Only the `Add` part is custom for each  component but rest of the code (`Save Changes`) is same everywhere.

Reply:
IMO, keeping it sep is more clean ignoring the duplication of "Save Changes"